llmfoundry/models/layers/blocks.py

```diff
@@ -132,6 +135,22 @@ def forward(
         m = x
         if self.norm_2 is not None:
             m = self.norm_2(x)
+        batch_size = m.size(0)
+        seq_len = m.size(1)
+        if not self.use_pad_tok_in_ffwd:
+            try:
+                from flash_attn.bert_padding import unpad_input  # type: ignore # yapf: disable # isort: skip
+            except:
+                raise RuntimeError(
+                    'Please install flash-attn==1.0.9 or flash-attn==2.3.2')
+            m, indices, _, _ = unpad_input(m, attention_mask)
         n = self.ffn(m)
+        if not self.use_pad_tok_in_ffwd:
+            try:
+                from flash_attn.bert_padding import pad_input  # type: ignore # yapf: disable # isort: skip
+            except:
+                raise RuntimeError(
+                    'Please install flash-attn==1.0.9 or flash-attn==2.3.2')
+            n = pad_input(n, indices, batch_size, seq_len)
```

Comment:
I'd put
```
try:
    from flash_attn.bert_padding import unpad_input, pad_input  # type: ignore # yapf: disable # isort: skip
except:
    unpad_input, pad_input = None, None
```
at the top of the file

then have this as
```suggestion
        batch_size = m.size(0)
        seq_len = m.size(1)
        if not self.use_pad_tok_in_ffwd:
            if unpad_input is None:
                raise RuntimeError(
                    'Please install flash-attn==1.0.9 or flash-attn==2.3.2')
            m, indices, _, _ = unpad_input(m, attention_mask)
        n = self.ffn(m)
        if not self.use_pad_tok_in_ffwd:
            n = pad_input(n, indices, batch_size, seq_len)
```
(removed the second runtime error cuz if the first one passed, the second one should be fine)